core/api/views/projects.py

```diff
@@ -73,6 +78,53 @@ class MetaProjectListView(generics.ListAPIView):
     serializer_class = MetaProjectSerializer
 
 
+class MetaProjectMyaListView(generics.ListAPIView):
+    """
+    List meta projects available for MYA update.
+    """
+
+    permission_classes = [HasMetaProjectsViewAccess]
+    serializer_class = MetaProjecMyaSerializer
+    filterset_class = MetaProjectMyaFilter
+
+    def get_queryset(self):
+        result = (
+            MetaProject.objects.filter(
+                type=MetaProject.MetaProjectType.MYA,
+                projects__submission_status__name="Approved",
+            )
+            .exclude(
+                projects__status__name="Completed",
+            )
+            .distinct()
+        )
+        return result
+
+
+class MetaProjectMyaDetailsViewSet(
+    viewsets.GenericViewSet,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+):
+    permission_classes = [HasMetaProjectsViewAccess]
```

Comment:
We should allow the update only for a permission with editing rights. I'm not sure who should edit what in this case. Maybe discuss with Miruna?